### PR TITLE
OMERO build documentation review

### PR DESCRIPTION
**Pull request by sbesson**
As part of the decoupling effort, this PR is a first attempt at re-organizing, extending and improving the documentation of the build system. The PR describes the current state of the build system. Another PR  will be opened later highlighting its changes specific to the Bio-Formats decoupling effort.

This work is still in progress but feedback on the current state is welcome.
- [x] Ivy configurations, 
- [x] add paragraph on lifecycle
- [x] describe caches
- [x] describe main resolver

/cc @bramalingam @hflynn @jburel

### CODE REVIEW

File: omero/developers/build-system.txt [hunk 1]
```diff
@@ -165,6 +87,81 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
...
+:ivydoc:`local file system <resolver/filesystem.html>`.
+
+Currently, the default resolver used by the build system, called
+``omero-resolver``, looks in the following locations by order of precedence:
+
+#. :file:`target/repository`
+#. :file:`lib/repository`
+#. :file:`target/test-repository`
+#. :file:`~/.m2/repository`
+#. `Maven central repository <http://search.maven.org>`_
```
Comment (hflynn) [inline omero/developers/build-system.txt]: Annoyingly, although this link works, it is yet another one that Jenkins is convinced doesn't work :( Can we do something to the link checker in the build to stop this issue?

File: omero/developers/build-system.txt [hunk 2]
```diff
@@ -5,157 +5,79 @@ Build System
...
-        |    |-- resources                      Other files of interest
-        |    |-- test                           Test source code and test resources
-        |    \-- target                         Output of build (deleted on clean)
+        |  NOTABLE COMPONENTS
         |
-        |-- model ............................. The model component is special in that in produces
-        |                                       a jar specific to your database choice: model-psql.jar
-        |                                       The generated `ome.model.*` files contain Hibernate
-        |                                       annotations for object-relational mapping.
+        |-- model ............................ The model component is special in that in produces
```
Comment (hflynn) [inline omero/developers/build-system.txt]: Typo -> 'in that in' - should be 'in that it'

File: omero/developers/build-system.txt [hunk 3]
```diff
@@ -165,6 +87,81 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
...
 directories named "generated".
 
+Build tools
+-----------
+
+Ant
+^^^
+
+``./build.py`` is a complete replacement for your local ant install. In
+many cases on and on most OS, you will be fine running :program:`ant`. If you
```
Comment (hflynn) [inline omero/developers/build-system.txt]: 'In many cases on and on most OS' is a bit scrambled. Suggest 'In most cases, you will...'

File: omero/developers/build-system.txt [hunk 4]
```diff
@@ -175,34 +172,66 @@ products which get built. Namely, the builds for the non-Java components
...
-not mix calls between the two.**
+Server build
+------------
+
+The default ant target (``build-default``) will build the OMERO system and
+copy the necessary components for a binary distribution to the :file:`dist`
+directory. Below is a comparison of what is taken from the build, where
+it is put, and what role it plays in the distribution.
+
+.. list-table::
```
Comment (hflynn) [inline omero/developers/build-system.txt]: Why is the first 'lib/server' using file mark-up when none of the subsequent items in this column are?

File: omero/developers/build-system.txt [hunk 5]
```diff
@@ -165,6 +87,81 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
...
+
+#. :file:`target/repository`
+#. :file:`lib/repository`
+#. :file:`target/test-repository`
+#. :file:`~/.m2/repository`
+#. `Maven central repository <http://search.maven.org>`_
+#. `OME artifactory <http://artifacts.openmicroscopy.org>`_
+#. `SpringSource repository <http://repository.springsource.com/>`_
+
+Note that the Maven Central and OME artifactory resolvers currently have
```
Comment (hflynn) [inline omero/developers/build-system.txt]: Note box?

File: omero/developers/build-system.txt [hunk 6]
```diff
@@ -5,157 +5,79 @@ Build System
...
-Build tools
------------
-
-OMERO mostly uses an `ant <http://ant.apache.org>`_-based build with
-dependency management provided by `Ivy <http://ant.apache.org/ivy>`_.
-:doc:`Native code </developers/Cpp>` is built using
-`SCons <http://scons.org>`__ and Python uses the traditional
-distutils/setuptools tools.
+OMERO mostly uses an Ant_-based build with dependency management provided by
+Ivy_. :doc:`Native code </developers/Cpp>` is built using Cmake and Python
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: Given that the only (?) native code is C++ would the link here better be 'Cpp code', say?

File: omero/developers/build-system.txt [hunk 7]
```diff
@@ -5,157 +5,79 @@ Build System
...
+        |                                      annotations for object-relational mapping.
         |
-        |-- blitz ............................. The blitz component also performs code generation
-        |    |                                  producing artifacts for Java, Python, and C++.
-        |    |
-        |    \ blitz_tools.py ................. OMERO-specific SCons Environment definition
-        |                                       and other build tools.
+        |-- blitz ............................ The blitz component also performs code generation
+        |    |                                 producing artifacts for Java, Python, and C++.
+        |    \ blitz_tools.py ................ OMERO-specific SCons Environment definition
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: Given the above chnage from SCons to Cmake is the mention of SCons here still relevant?

File: omero/developers/build-system.txt [hunk 8]
```diff
@@ -5,157 +5,79 @@ Build System
...
-
-::
-
-    ./build.py -f components/server/build.xml
-
-Code generation
-^^^^^^^^^^^^^^^
-
 Unfortunately, just the above snapshot of the code repository omits some
 of the most important code. Many MB of source code is generated both by
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: It's not part of these chnages but I think 'many megabytes' is much more readable.

File: omero/developers/build-system.txt [hunk 9]
```diff
@@ -165,6 +87,106 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
...
+#. a local cache under :file:`lib/cache`
+#. the maven cache under :file:`~/.m2/repository`
+
+.. note::
+
+   When the Ivy configuration file or the version number is changed, the
+   cache can become stale. Calling ``./build.py clean`` from the top-level
+   build will delete the content of the local cache.
+
+:ivydoc:`Resolvers <settings/resolvers.html>` are key to how Ivy functions. Multi-resolvers can be defined (as granular as for an individual jar) in order
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: What is meant by 'Multi-resolvers' here? Do you mean 'Multiple resolvers'? And by 'as granular' are you saying fine-grained enough to resolve an individual jar?

File: omero/developers/build-system.txt [hunk 10]
```diff
@@ -165,6 +87,106 @@ of the :model_doc:`OME-Model <ome-xml/>` and generate our |OmeroModel|.
...
+#. install: convert the component Ivy file into a pom file using
+   :ivydoc:`makepom <use/makepom.html>` and
+   :ivydoc:`publish <use/publish.html>` the component artifacts
+
+Individual components can override the content of this default lifecycle via
+their :file:`build.xml`.
+
 .. _build#OmeroTools:
 
 OmeroTools
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: Just below this line...

"Similarly' to what? This section may have relied on an earlier section being directly above it. I'd just drop the word and start, 'The ant build...'

File: omero/developers/build-system.txt [hunk 11]
```diff
@@ -175,34 +197,63 @@ products which get built. Namely, the builds for the non-Java components
...
+    * - etc/
+      - :file:`etc/`
+      - Configuration
+    * - :file:`sql/*.sql`
+      - :file:`sql/`
+      - SQL scripts to prepare the database
+
+
+.. note::
+    By default, |OmeroCpp| is not built. Use ``build-all`` for that.
```
Comment (ximenesuk) [inline omero/developers/build-system.txt]: 'are built' Once the reference OmeroCpp is expanded it is a plural.

**Comment by ximenesuk** [comment]
All the chnages look good to me.

**Comment by hflynn** [comment]
Are you done with this PR @sbesson ?

**Comment by sbesson** [comment]
Yes. The rest of the developer documentation can be opened once the BF decoupling is merged.

**Pull request merged**